src/threading.c

```diff
@@ -816,10 +842,12 @@ void _jl_mutex_wait(jl_task_t *self, jl_mutex_t *lock, int safepoint)
         return;
     }
     JL_TIMING(LOCK_SPIN, LOCK_SPIN);
+    uint64_t t0 = jl_hrtime();
     while (1) {
         if (owner == NULL && jl_atomic_cmpswap(&lock->owner, &owner, self)) {
             lock->count = 1;
             jl_profile_lock_acquired(lock);
+            jl_record_lock_spin_time(jl_hrtime() - t0);
```

Comment:
currently time in runtime-internal spin locks and Julia-side `SpinLock`s accumulate into the same field... idk if we want to separate those (i guess some use of Julia-side `SpinLock`s are "internal" not just in user-code, so i'm leaning towards keeping accumulating them both into the same field)